1.19.0-alpha.4

matteobruni opened the pull request:
refactor: refactored inferno/preact/react component to use tsParticles instead Container
fix: fixes undefined on TextDrawer (feat. #1087)
refactor: moved Mover from particle to particles
refactor: refactored NumberUtils, now wraps the exported function like ColorUtils
refactor: refactored Utils, static methods are now exported functions
refactor: refactored CanvasUtils, static methods are now exported functions
refactor: particles velocity vector uses a class that handles correctly all the speed/angle calculations (feat. #1093). Particle velocity could be different after this fix
refactor: values that could be a range now use the new `RangeValue` type (`number | { min: number, max: number }`), this replaces the `random.minimumValue` and `animation.minimumValue`, this is a non-breaking change, old values are still supported (feat. #1101)

matteobruni committed:
fix: fixed color animations

ghost commented:
## DeepCode's analysis on [#e7b552](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5facb7357e395042093108ee48de6229a65013ed/matteobruni/tsparticles/e7b552c224f35d73ebb5412c95515edda75131aa/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=3&) found:

- :warning: **1** warning, :information_source: **3** minor issues. :point_down:
- :heavy_check_mark: **3** issues were fixed.
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Function divModeExecute has no return statements and hence always returns undefined, but it is called here and its return value is used here. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Utils/Utils.ts#L572">Utils.ts:572</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Utils/Utils.ts#L576">Utils.ts:576</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Utils/Utils.ts#L588">Utils.ts:588</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5facb7357e395042093108ee48de6229a65013ed/matteobruni/tsparticles/e7b552c224f35d73ebb5412c95515edda75131aa/pr/_/%2Fcore%2Fmain%2Fsrc%2FUtils%2FUtils.ts/javascript%2Fdc_interfile_project%2FUsageOfUndefinedReturnValue/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">'public' is implicit. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Options/Classes/Particles/Move/MoveAngle.ts#L9">MoveAngle.ts:9</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Options/Classes/Particles/Move/MoveAngle.ts#L10">MoveAngle.ts:10</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Options/Classes/Particles/Move/MoveAngle.ts#L17">MoveAngle.ts:17</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5facb7357e395042093108ee48de6229a65013ed/matteobruni/tsparticles/e7b552c224f35d73ebb5412c95515edda75131aa/pr/_/%2Fcore%2Fmain%2Fsrc%2FOptions%2FClasses%2FParticles%2FMove%2FMoveAngle.ts/javascript%2Ftslint%2Fmember-access/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
    <tr>
      <td width="77%">Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead. Occurrences: <ul><li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Core/Particle.ts#L610">Particle.ts:610</a></li> <li><a href="https://github.com/matteobruni/tsparticles/blob/e7b552c224f35d73ebb5412c95515edda75131aa/core/main/src/Core/Particle.ts#L611">Particle.ts:611</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5facb7357e395042093108ee48de6229a65013ed/matteobruni/tsparticles/e7b552c224f35d73ebb5412c95515edda75131aa/pr/_/%2Fcore%2Fmain%2Fsrc%2FCore%2FParticle.ts/javascript%2Ftslint%2Farray-type/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/matteobruni/tsparticles/5facb7357e395042093108ee48de6229a65013ed/matteobruni/tsparticles/e7b552c224f35d73ebb5412c95515edda75131aa/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=3&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=matteobruni)

matteobruni committed:
refactor: life updater and life properties merged in an interface used by Particle

matteobruni committed:
fix: fixed particle initial position
build: fixed build

matteobruni committed:
refactor: particle spin properties unified in a single interface

matteobruni committed:
refactor: refactored inferno/preact/react component to use tsParticles instead Container
feat: added url to all component to load an external json config

matteobruni committed:
build: fixed tests

matteobruni committed:
build: fixed build

matteobruni committed:
build: fixed angular build

matteobruni committed:
build: fixed angular build

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.33.3 to 2.34.0

Bumps [rollup](https://github.com/rollup/rollup) from 2.33.3 to 2.34.0.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.33.3...v2.34.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1072 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.34.0

build(deps-dev): bump rollup from 2.33.3 to 2.34.0

matteobruni committed:
build: deps updated

matteobruni committed:
build: deps updated

dependabot[bot] committed:
build(deps-dev): bump webpack from 5.8.0 to 5.9.0

Bumps [webpack](https://github.com/webpack/webpack) from 5.8.0 to 5.9.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.8.0...v5.9.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump prettier from 2.2.0 to 2.2.1

Bumps [prettier](https://github.com/prettier/prettier) from 2.2.0 to 2.2.1.
- [Release notes](https://github.com/prettier/prettier/releases)
- [Changelog](https://github.com/prettier/prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/prettier/compare/2.2.0...2.2.1)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1073 from matteobruni/dependabot/npm_and_yarn/dev/webpack-5.9.0

build(deps-dev): bump webpack from 5.8.0 to 5.9.0

matteobruni committed:
Merge pull request #1074 from matteobruni/dependabot/npm_and_yarn/dev/prettier-2.2.1

build(deps-dev): bump prettier from 2.2.0 to 2.2.1

matteobruni committed:
build: updated deps

dependabot[bot] committed:
build(deps-dev): bump @rollup/plugin-commonjs from 16.0.0 to 17.0.0

Bumps [@rollup/plugin-commonjs](https://github.com/rollup/plugins) from 16.0.0 to 17.0.0.
- [Release notes](https://github.com/rollup/plugins/releases)
- [Commits](https://github.com/rollup/plugins/compare/commonjs-v16.0.0...commonjs-v17.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump webpack-bundle-analyzer from 4.1.0 to 4.2.0

Bumps [webpack-bundle-analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer) from 4.1.0 to 4.2.0.
- [Release notes](https://github.com/webpack-contrib/webpack-bundle-analyzer/releases)
- [Changelog](https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/webpack-bundle-analyzer/compare/v4.1.0...v4.2.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @rollup/plugin-typescript from 6.1.0 to 8.0.0

Bumps [@rollup/plugin-typescript](https://github.com/rollup/plugins) from 6.1.0 to 8.0.0.
- [Release notes](https://github.com/rollup/plugins/releases)
- [Commits](https://github.com/rollup/plugins/compare/typescript-v6.1.0...eslint-v8.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump babel-loader from 8.2.1 to 8.2.2

Bumps [babel-loader](https://github.com/babel/babel-loader) from 8.2.1 to 8.2.2.
- [Release notes](https://github.com/babel/babel-loader/releases)
- [Changelog](https://github.com/babel/babel-loader/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel-loader/compare/v8.2.1...v8.2.2)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump svelte from 3.30.0 to 3.30.1

Bumps [svelte](https://github.com/sveltejs/svelte) from 3.30.0 to 3.30.1.
- [Release notes](https://github.com/sveltejs/svelte/releases)
- [Changelog](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte/compare/v3.30.0...v3.30.1)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1078 from matteobruni/dependabot/npm_and_yarn/dev/webpack-bundle-analyzer-4.2.0

build(deps-dev): bump webpack-bundle-analyzer from 4.1.0 to 4.2.0

matteobruni committed:
Merge pull request #1079 from matteobruni/dependabot/npm_and_yarn/dev/rollup/plugin-typescript-8.0.0

build(deps-dev): bump @rollup/plugin-typescript from 6.1.0 to 8.0.0

matteobruni committed:
Merge pull request #1080 from matteobruni/dependabot/npm_and_yarn/dev/babel-loader-8.2.2

build(deps-dev): bump babel-loader from 8.2.1 to 8.2.2

matteobruni committed:
Merge pull request #1081 from matteobruni/dependabot/npm_and_yarn/dev/svelte-3.30.1

build(deps): bump svelte from 3.30.0 to 3.30.1

matteobruni committed:
Merge pull request #1077 from matteobruni/dependabot/npm_and_yarn/dev/rollup/plugin-commonjs-17.0.0

build(deps-dev): bump @rollup/plugin-commonjs from 16.0.0 to 17.0.0

matteobruni committed:
docs: improved some readmes

dependabot[bot] committed:
build(deps-dev): bump copyfiles from 2.4.0 to 2.4.1

Bumps [copyfiles](https://github.com/calvinmetcalf/copyfiles) from 2.4.0 to 2.4.1.
- [Release notes](https://github.com/calvinmetcalf/copyfiles/releases)
- [Commits](https://github.com/calvinmetcalf/copyfiles/compare/v2.4.0...v2.4.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @rollup/plugin-node-resolve from 10.0.0 to 11.0.0

Bumps [@rollup/plugin-node-resolve](https://github.com/rollup/plugins) from 10.0.0 to 11.0.0.
- [Release notes](https://github.com/rollup/plugins/releases)
- [Commits](https://github.com/rollup/plugins/compare/node-resolve-v10.0.0...commonjs-v11.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump uglify-js from 3.12.0 to 3.12.1

Bumps [uglify-js](https://github.com/mishoo/UglifyJS) from 3.12.0 to 3.12.1.
- [Release notes](https://github.com/mishoo/UglifyJS/releases)
- [Commits](https://github.com/mishoo/UglifyJS/compare/v3.12.0...v3.12.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 4.8.2 to 4.9.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.9.0/packages/eslint-plugin)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1082 from matteobruni/dependabot/npm_and_yarn/dev/copyfiles-2.4.1

build(deps-dev): bump copyfiles from 2.4.0 to 2.4.1

matteobruni committed:
Merge pull request #1083 from matteobruni/dependabot/npm_and_yarn/dev/rollup/plugin-node-resolve-11.0.0

build(deps-dev): bump @rollup/plugin-node-resolve from 10.0.0 to 11.0.0

matteobruni committed:
Merge pull request #1084 from matteobruni/dependabot/npm_and_yarn/dev/uglify-js-3.12.1

build(deps-dev): bump uglify-js from 3.12.0 to 3.12.1

matteobruni committed:
Merge pull request #1086 from matteobruni/dependabot/npm_and_yarn/dev/typescript-eslint/eslint-plugin-4.9.0

build(deps-dev): bump @typescript-eslint/eslint-plugin from 4.8.2 to 4.9.0

matteobruni committed:
build: updated deps

matteobruni committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	demo/react/yarn.lock
#	demo/svelte/yarn.lock
#	demo/vue/yarn.lock

matteobruni committed:
build: updated deps

matteobruni committed:
fix: fixes undefined on TextDrawer, closes #1087

matteobruni committed:
docs: added jquery awesome badge

dependabot[bot] committed:
build(deps-dev): bump postcss from 8.1.10 to 8.1.13

Bumps [postcss](https://github.com/postcss/postcss) from 8.1.10 to 8.1.13.
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.1.10...8.1.13)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @babel/core from 7.12.3 to 7.12.9

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.12.3 to 7.12.9.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.12.9/packages/babel-core)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1091 from matteobruni/dependabot/npm_and_yarn/dev/postcss-8.1.13

build(deps-dev): bump postcss from 8.1.10 to 8.1.13

matteobruni committed:
Merge pull request #1092 from matteobruni/dependabot/npm_and_yarn/dev/babel/core-7.12.9

build(deps-dev): bump @babel/core from 7.12.3 to 7.12.9

matteobruni committed:
build: updated deps

dependabot[bot] committed:
build(deps-dev): bump @angular-devkit/build-angular

Bumps [@angular-devkit/build-angular](https://github.com/angular/angular-cli) from 0.1100.2 to 0.1100.3.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/commits)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
refactor: optimized particles sorting

matteobruni committed:
Merge pull request #1090 from matteobruni/dependabot/npm_and_yarn/dev/angular-devkit/build-angular-0.1100.3

build(deps-dev): bump @angular-devkit/build-angular from 0.1100.2 to 0.1100.3

matteobruni committed:
refactor: moved Mover from particle to particles

matteobruni committed:
refactor: refactored NumberUtils, now wraps the exported function like ColorUtils

matteobruni committed:
refactor: refactored Utils, static methods are now exported functions

matteobruni committed:
build: fixed build and tests

matteobruni committed:
refactor: refactored CanvasUtils, static methods are now exported functions

matteobruni committed:
build: cleaned imports

matteobruni committed:
fix: fixed previous sorting improvements

matteobruni committed:
feat: starting to work with speed/angle instead of horizontal/vertical

dependabot[bot] committed:
build(deps-dev): bump svelte-check from 1.1.17 to 1.1.18

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.1.17 to 1.1.18.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/commits)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump ts-node from 9.0.0 to 9.1.0

Bumps [ts-node](https://github.com/TypeStrong/ts-node) from 9.0.0 to 9.1.0.
- [Release notes](https://github.com/TypeStrong/ts-node/releases)
- [Commits](https://github.com/TypeStrong/ts-node/compare/v9.0.0...v9.1.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/animations from 11.0.2 to 11.0.3

Bumps [@angular/animations](https://github.com/angular/angular/tree/HEAD/packages/animations) from 11.0.2 to 11.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/11.0.3/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/11.0.3/packages/animations)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @angular/compiler-cli from 11.0.2 to 11.0.3

Bumps [@angular/compiler-cli](https://github.com/angular/angular/tree/HEAD/packages/compiler-cli) from 11.0.2 to 11.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/11.0.3/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/11.0.3/packages/compiler-cli)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1095 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.1.18

build(deps-dev): bump svelte-check from 1.1.17 to 1.1.18

matteobruni committed:
Merge pull request #1097 from matteobruni/dependabot/npm_and_yarn/dev/ts-node-9.1.0

build(deps-dev): bump ts-node from 9.0.0 to 9.1.0

matteobruni committed:
Merge pull request #1098 from matteobruni/dependabot/npm_and_yarn/dev/angular/animations-11.0.3

build(deps): bump @angular/animations from 11.0.2 to 11.0.3

matteobruni committed:
Merge pull request #1099 from matteobruni/dependabot/npm_and_yarn/dev/angular/compiler-cli-11.0.3

build(deps-dev): bump @angular/compiler-cli from 11.0.2 to 11.0.3

dependabot[bot] committed:
build(deps): bump svelte from 3.30.1 to 3.31.0

Bumps [svelte](https://github.com/sveltejs/svelte) from 3.30.1 to 3.31.0.
- [Release notes](https://github.com/sveltejs/svelte/releases)
- [Changelog](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sveltejs/svelte/compare/v3.30.1...v3.31.0)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1096 from matteobruni/dependabot/npm_and_yarn/dev/svelte-3.31.0

build(deps): bump svelte from 3.30.1 to 3.31.0

matteobruni committed:
build: used a vector class for velocity to use angle/length and horizontal/vertical

matteobruni committed:
build: used Velocity class in more places, added length/angle to IVelocity
fix: fixed the move angle option

matteobruni committed:
fix: fixed color animation loop
refactor: used some new velocity class methods

matteobruni committed:
docs: added docs for issue #1100
refactor: changed more velocity code using the new Velocity class

matteobruni committed:
fix: changed the move particle method reducing the speed to quarter instead of half

matteobruni committed:
refactor: removed the IVelocity interface, the new Velocity class has replaced it

matteobruni committed:
build: removed an unused function in tests

matteobruni committed:
build: some velocity class refactoring

dependabot[bot] committed:
build(deps): bump @angular/platform-browser from 11.0.2 to 11.0.3

Bumps [@angular/platform-browser](https://github.com/angular/angular/tree/HEAD/packages/platform-browser) from 11.0.2 to 11.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/11.0.3/packages/platform-browser)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-plugin-prettier from 3.1.4 to 3.2.0

Bumps [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) from 3.1.4 to 3.2.0.
- [Release notes](https://github.com/prettier/eslint-plugin-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-plugin-prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-plugin-prettier/compare/v3.1.4...v3.2.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump eslint-config-prettier from 6.15.0 to 7.0.0

Bumps [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) from 6.15.0 to 7.0.0.
- [Release notes](https://github.com/prettier/eslint-config-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-config-prettier/compare/v6.15.0...v7.0.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump core-js from 3.8.0 to 3.8.1

Bumps [core-js](https://github.com/zloirock/core-js) from 3.8.0 to 3.8.1.
- [Release notes](https://github.com/zloirock/core-js/releases)
- [Changelog](https://github.com/zloirock/core-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/zloirock/core-js/compare/v3.8.0...v3.8.1)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump svelte-check from 1.1.18 to 1.1.19

Bumps [svelte-check](https://github.com/sveltejs/language-tools) from 1.1.18 to 1.1.19.
- [Release notes](https://github.com/sveltejs/language-tools/releases)
- [Commits](https://github.com/sveltejs/language-tools/commits)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1106 from matteobruni/dependabot/npm_and_yarn/dev/svelte-check-1.1.19

build(deps-dev): bump svelte-check from 1.1.18 to 1.1.19

matteobruni committed:
Merge pull request #1105 from matteobruni/dependabot/npm_and_yarn/dev/core-js-3.8.1

build(deps): bump core-js from 3.8.0 to 3.8.1

matteobruni committed:
Merge pull request #1103 from matteobruni/dependabot/npm_and_yarn/dev/eslint-plugin-prettier-3.2.0

build(deps-dev): bump eslint-plugin-prettier from 3.1.4 to 3.2.0

matteobruni committed:
Merge pull request #1104 from matteobruni/dependabot/npm_and_yarn/dev/eslint-config-prettier-7.0.0

build(deps-dev): bump eslint-config-prettier from 6.15.0 to 7.0.0

matteobruni committed:
Merge pull request #1102 from matteobruni/dependabot/npm_and_yarn/dev/angular/platform-browser-11.0.3

build(deps): bump @angular/platform-browser from 11.0.2 to 11.0.3

matteobruni committed:
Merge pull request #1094 from matteobruni/feat-1093

Feature #1093

matteobruni committed:
refactor: created Range type, first refactoring using the new Range type

matteobruni committed:
build: fixed an import

matteobruni committed:
refactor: square operations refactored with ** operator

matteobruni committed:
build: fixed tests

matteobruni committed:
refactor: improved some Range accesses

matteobruni committed:
refactor: improved some Range accesses

dependabot[bot] committed:
build(deps-dev): bump source-map-loader from 1.1.2 to 1.1.3

Bumps [source-map-loader](https://github.com/webpack-contrib/source-map-loader) from 1.1.2 to 1.1.3.
- [Release notes](https://github.com/webpack-contrib/source-map-loader/releases)
- [Changelog](https://github.com/webpack-contrib/source-map-loader/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack-contrib/source-map-loader/compare/v1.1.2...v1.1.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump typescript-json-schema from 0.44.1 to 0.45.0

Bumps [typescript-json-schema](https://github.com/YousefED/typescript-json-schema) from 0.44.1 to 0.45.0.
- [Release notes](https://github.com/YousefED/typescript-json-schema/releases)
- [Commits](https://github.com/YousefED/typescript-json-schema/compare/v0.44.1...v0.45.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump @angular/common from 11.0.2 to 11.0.3

Bumps [@angular/common](https://github.com/angular/angular/tree/HEAD/packages/common) from 11.0.2 to 11.0.3.
- [Release notes](https://github.com/angular/angular/releases)
- [Changelog](https://github.com/angular/angular/blob/master/CHANGELOG.md)
- [Commits](https://github.com/angular/angular/commits/11.0.3/packages/common)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump @angular/cli from 11.0.2 to 11.0.3

Bumps [@angular/cli](https://github.com/angular/angular-cli) from 11.0.2 to 11.0.3.
- [Release notes](https://github.com/angular/angular-cli/releases)
- [Commits](https://github.com/angular/angular-cli/compare/v11.0.2...v11.0.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps-dev): bump rollup from 2.34.0 to 2.34.2

Bumps [rollup](https://github.com/rollup/rollup) from 2.34.0 to 2.34.2.
- [Release notes](https://github.com/rollup/rollup/releases)
- [Changelog](https://github.com/rollup/rollup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rollup/rollup/compare/v2.34.0...v2.34.2)

Signed-off-by: dependabot[bot] <support@github.com>

matteobruni committed:
Merge pull request #1109 from matteobruni/dependabot/npm_and_yarn/dev/source-map-loader-1.1.3

build(deps-dev): bump source-map-loader from 1.1.2 to 1.1.3

matteobruni committed:
Merge pull request #1110 from matteobruni/dependabot/npm_and_yarn/dev/typescript-json-schema-0.45.0

build(deps-dev): bump typescript-json-schema from 0.44.1 to 0.45.0

matteobruni committed:
Merge pull request #1111 from matteobruni/dependabot/npm_and_yarn/dev/angular/common-11.0.3

build(deps): bump @angular/common from 11.0.2 to 11.0.3

matteobruni committed:
Merge pull request #1112 from matteobruni/dependabot/npm_and_yarn/dev/angular/cli-11.0.3

build(deps-dev): bump @angular/cli from 11.0.2 to 11.0.3

matteobruni committed:
Merge pull request #1113 from matteobruni/dependabot/npm_and_yarn/dev/rollup-2.34.2

build(deps-dev): bump rollup from 2.34.0 to 2.34.2

matteobruni committed:
refactor: more RangeValue usage, added to movement speed too

matteobruni committed:
build: fixed tests

matteobruni committed:
build: simplified setRangeValue

matteobruni committed:
Merge pull request #1107 from matteobruni/feat-1101

Feat #1101

matteobruni committed:
build: updated deps

matteobruni committed:
build: prettied code

matteobruni committed:
chore(release): published new version

 - ng-particles@2.2.0-alpha.4
 - inferno-particles@1.2.0-alpha.4
 - jquery-particles@1.19.0-alpha.4
 - preact-particles@1.19.0-alpha.4
 - react-tsparticles@1.19.0-alpha.4
 - svelte-particles@1.2.0-alpha.4
 - particles.vue@2.2.0-alpha.4
 - particles.vue3@1.2.0-alpha.4
 - tsparticles-editor@1.19.0-alpha.4
 - tsparticles@1.19.0-alpha.4
 - ng-particles-demo@1.4.0-alpha.4
 - editor-demo@1.4.0-alpha.4
 - inferno-particles-demo@1.2.0-alpha.4
 - jquery-particles-demo@1.4.0-alpha.4
 - tsparticles-demo@1.4.0-alpha.4
 - preact-particles-demo@1.4.0-alpha.4
 - react-tsparticles-demo@1.4.0-alpha.4
 - svelte-demo@1.2.0-alpha.4
 - particles.vue-demo@1.4.0-alpha.4
 - particles.vue3-demo@1.2.0-alpha.4
 - tsparticles-preset-big-circles@1.4.0-alpha.4
 - tsparticles-preset-fire@1.4.0-alpha.4
 - tsparticles-preset-fountain@1.4.0-alpha.4
 - tsparticles-preset-links@1.4.0-alpha.4
 - tsparticles-preset-sea-anemone@1.4.0-alpha.4
 - tsparticles-preset-snow@1.4.0-alpha.4
 - tsparticles-preset-stars@1.4.0-alpha.4
 - tsparticles-shape-bubble@1.4.0-alpha.4
 - tsparticles-shape-heart@1.4.0-alpha.4
 - tsparticles-shape-multiline-text@1.4.0-alpha.4
 - tsparticles-shape-spiral@1.4.0-alpha.4
 - cra-template-particles-typescript@1.2.0-alpha.4
 - cra-template-particles@1.2.0-alpha.4
